src/jlapi.c

```diff
@@ -36,10 +40,54 @@ DLLEXPORT void jl_init_with_image(const char *julia_home_dir, const char *image_
 {
     if (jl_is_initialized()) return;
     libsupport_init();
-    jl_compileropts.julia_home = julia_home_dir;
+
+    // set jl_compileropts.julia_home
+    char *lib_path = (char*)malloc(PATH_MAX);
+    if (julia_home_dir != NULL) {
```

Comment:
this code should go in jl_resolve_sysimg_location rather than duplicating part of that code here. https://github.com/JuliaLang/julia/blob/0181b0e2626ede655e85eb0cb61d10957a82dd3d/src/init.c#L854